### PR TITLE
Streamline init container config

### PR DESCRIPTION
**Pull request by luhi-DT**
## Description

We should only have 1 way to configure each thing, so the same thing can't be in env and the secret.
The general rule is that the env should be used for things that are only available at the time of injection.

What was moved?
- InstallerMode
- CSIMode
- ReadOnlyCSI

## How can this be tested?

Everything should work as before, deploy a Dynakube with applicationMonitoring and test -> readOnly CSI, set a fixed version and a dynakube without readOnlyCSI but CSI still enabled

## Checklist

- [x] Unit tests have been updated/added
- [x] PR is labeled accordingly with a single label
- [x] I have read and understood the [contribution guidelines](https://github.com/Dynatrace/dynatrace-operator/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Review by andriisoldatenko — APPROVED** [review]
code LGTM.

File: pkg/injection/startup/secret.go [hunk 1]
```diff
@@ -29,6 +29,8 @@ type SecretConfig struct {
 	HostGroup           string            `json:"hostGroup"`
 	InitialConnectRetry int               `json:"initialConnectRetry"`
 	EnforcementMode     bool              `json:"enforcementMode"`
+	CSIMode             bool              `json:"CSIMode"`
```
Comment (waodim) [inline pkg/injection/startup/secret.go:32]:
```suggestion
	CSIMode             bool              `json:"csiMode"`
```

Comment (waodim) [reply pkg/injection/startup/secret.go:32]: apart from this suggestion it also LGTM.

Comment (luhi-DT) [reply pkg/injection/startup/secret.go]: good catch

**Review by waodim — APPROVED** [review]
LGTM

**Pull request merged**